### PR TITLE
add app offset to debug adapter (VSC-492)

### LINKED ISSUE #225: [Feature Request]: Use variable app_image_offset for debugAdapter (VSC-490)
When debugging Rainmaker apps, the debugAdapter fails because the app partition used for them is at addr: `0x20000` instead of `0x10000`, the default. The debugAdapter CLI has an option for it: 
```
Usage: debug_adapter_main.py [OPTIONS]

Options:
  -a, --app_flash_off INTEGER     Program start address offset
                                  (ESP32_APP_FLASH_OFF)  [default: 65536]
...
```
the esp_debug_adapter file `esp.py` in the hw_specific folder hardcodes `0x10000`
Line 114: `       self.app_flash_offset = 0x10000  # default for for ESP xtensa chips`

I figured maybe the app partition can be parsed from the flasher_args.json and added in this area:
https://github.com/espressif/vscode-esp-idf-extension/blob/6eddcb8917b1a7c3885b7c66dcc97f6f2ee3373e/src/espIdf/debugAdapter/debugAdapterManager.ts#L108-L120

I don't really know if it can all be done in the vscode-esp-idf-extension repo or if the esp-debug-adapter also needs to be changed.
For now, I just edit the `esp.py` file (line 114) when debugging Rainmaker apps, but it's not very convenient.

Thanks though for all your hard work!

### PR DESCRIPTION
**Pull request by brianignacio5**
Previously the app offset was not being given to debug adapter parameters.

Now, parsing `your_project_directory/build/flasher_args.json` for app offset and include it to debug adapter arguments.

Fix #225

### CODE REVIEW

**Comment by meltdown03** [comment]
I tried PR but get this error:
```
Error: Invalid value for '--app_flash_off' / '-a': 0x20000 is not a valid integer
[Stopped] : ESP-IDF Debug Adapter
```

File: src/espIdf/debugAdapter/debugAdapterManager.ts [hunk 1]
```diff
@@ -117,6 +132,8 @@ export class DebugAdapterManager extends EventEmitter {
         this.port.toString(),
         "-dn",
         this.target,
+        "-a",
+        model.app.address,
```
Comment (meltdown03) [inline src/espIdf/debugAdapter/debugAdapterManager.ts:136]: I got a little further by changing this to:
`parseInt(model.app.address, 16).toString(),`
but Gdb is still using 0x10000 as shown in the debug log

**Comment by meltdown03** [comment]
I got further using this:
```
this.target,
        "-a",
        parseInt(model.app.address, 16).toString(),
      ];
```
but here is the log showing GDB still running `mon esp appimage_offset 0x10000`:
[debug.log](https://github.com/espressif/vscode-esp-idf-extension/files/5650117/debug.log)

**Comment by brianignacio5** [comment]
Hi @meltdown03 Thanks for trying this out, although was not finished yet hehe. 

Will keep testing and update soon !

File: src/espIdf/debugAdapter/debugAdapterManager.ts (hunk 1 above)
Comment (brianignacio5) [reply src/espIdf/debugAdapter/debugAdapterManager.ts]: > Here is the log showing GDB still running `mon esp appimage_offset 0x10000` command:
> [debug.log](https://github.com/espressif/vscode-esp-idf-extension/files/5650117/debug.log)
> EDIT:
> I found a workaround for the GDB issue. In launch.json, I set:
> 
> ```
> {
>             "type": "espidf",
>             "name": "Launch",
>             "request": "launch",
>             "logLevel": 5,
>             "gdbinitFile": "${workspaceFolder}/build/gdbinit",
> 
>         },
> ```
> 
> gdbinit:
> 
> ```
> target remote :3333
> symbol-file /home/neal/esp/esp-rainmaker/examples/led_light/build/led_light.elf
> mon reset halt
> flushregs
> thb app_main
> c
> ```
> 
> I can then debug without error. I still have that line edited though with `parseInt`

If `gdbinit` file is not specified, then https://github.com/espressif/esp-debug-adapter/blob/5439cd3c6e5632d8fef6754b59d2d3820f335854/debug_adapter/debug_backend/debug_backend/hw_specific/esp_init.gdb is used. Need to look into it.

Comment (brianignacio5) [reply src/espIdf/debugAdapter/debugAdapterManager.ts]: Can you share the elf file ?

**Comment by an-dr** [comment]
@meltdown03 I have found a bug in the [esp-debug-adapter](https://github.com/espressif/esp-debug-adapter) will fix it soon, thank you

File: src/espIdf/debugAdapter/debugAdapterManager.ts (hunk 1 above)
Comment (meltdown03) [reply src/espIdf/debugAdapter/debugAdapterManager.ts]: Still want the led_light.elf file?

**Comment by meltdown03** [comment]
> @meltdown03 I have found a bug in the [esp-debug-adapter](https://github.com/espressif/esp-debug-adapter). Will fix it soon, thank you

Cool, thanks, any chance you can tell me what it is so i can change it locally? Unless it requires a lot of changes...

**Comment by brianignacio5** [comment]
PTAL @pwmb @meltdown03 @gerekon

**Comment by meltdown03** [comment]
> PTAL @pwmb @meltdown03 @gerekon

Works good for me, finds the app_offset automatically or I can set it manually in `launch.json`.

**Review by pwmb — APPROVED** [review]

**Pull request merged**